x-pack/plugins/cases/server/client/cases/create.ts

```diff
@@ -67,6 +71,13 @@ export const create = async ({
   );
 
   try {
+    try {
+      await auth.ensureAuthorized(query.class, WriteOperations.Create);
+    } catch (error) {
+      // TODO: log error using audit logger
```

Comment:
I think is best to move the responsibility of logging to the `ensureAuthorized` function. Otherwise, we have to try/catch on every CRUD path. Also, if in the future, we need to change something in the log we will need to it on every CRUD path.

Reply:
Yeah, I was following the way the alerting team did it. It looks like the do an audit log within the `ensureAuthorized` and after each call to `ensureAuthorized`: https://github.com/elastic/kibana/blob/master/x-pack/plugins/alerting/server/alerts_client/alerts_client.ts#L241

https://github.com/elastic/kibana/blob/628bb4b3775aba5fee822e76fc8152ccc6866087/x-pack/plugins/alerting/server/authorization/alerts_authorization.ts#L164

Reply:
I see. Let's talk about it when we starting logging messages.